src/ophyd_async/fastcs/panda/_table.py

```diff
@@ -34,137 +36,88 @@ class SeqTrigger(str, Enum):
-    outc2: Optional[npt.NDArray[np.bool_]] = None,
-    outd2: Optional[npt.NDArray[np.bool_]] = None,
-    oute2: Optional[npt.NDArray[np.bool_]] = None,
-    outf2: Optional[npt.NDArray[np.bool_]] = None,
-) -> SeqTable:
-    """
-    Constructs a sequence table from a series of columns as arrays.
-    time2 is the only required argument and must not be None.
-    All other provided arguments must be of equal length to time2.
-    If any other argument is not given, or else given as None or empty,
-    an array of length len(time2) filled with the following is defaulted:
-    repeats: 1
-    trigger: SeqTrigger.IMMEDIATE
-    all others: 0/False as appropriate
-    """
-    assert time2 is not None, "time2 must be provided"
-    length = len(time2)
-    assert 0 < length < 4096, f"Length {length} not in range"
-
-    def or_default(
-        value: Optional[npt.NDArray[T]], dtype: Type[T], default_value: int = 0
-    ) -> npt.NDArray[T]:
-        if value is None or len(value) == 0:
-            return np.full(length, default_value, dtype=dtype)
-        return value
-
-    table = SeqTable(
-        repeats=or_default(repeats, np.uint16, 1),
-        trigger=trigger or [SeqTrigger.IMMEDIATE] * length,
-        position=or_default(position, np.int32),
-        time1=or_default(time1, np.uint32),
-        outa1=or_default(outa1, np.bool_),
-        outb1=or_default(outb1, np.bool_),
-        outc1=or_default(outc1, np.bool_),
-        outd1=or_default(outd1, np.bool_),
-        oute1=or_default(oute1, np.bool_),
-        outf1=or_default(outf1, np.bool_),
-        time2=time2,
-        outa2=or_default(outa2, np.bool_),
-        outb2=or_default(outb2, np.bool_),
-        outc2=or_default(outc2, np.bool_),
-        outd2=or_default(outd2, np.bool_),
-        oute2=or_default(oute2, np.bool_),
-        outf2=or_default(outf2, np.bool_),
-    )
-    for k, v in table.items():
-        size = len(v)  # type: ignore
-        if size != length:
-            raise ValueError(f"{k}: has length {size} not {length}")
-    return table
+PydanticNp1DArrayInt32 = Annotated[
+    np.ndarray[tuple[int], np.int32],
+    NpArrayPydanticAnnotation.factory(
+        data_type=np.int32, dimensions=1, strict_data_typing=False
+    ),
+    Field(default_factory=lambda: np.array([], np.int32)),
+]
+PydanticNp1DArrayBool = Annotated[
+    np.ndarray[tuple[int], np.bool_],
+    NpArrayPydanticAnnotation.factory(
+        data_type=np.bool_, dimensions=1, strict_data_typing=False
+    ),
+    Field(default_factory=lambda: np.array([], dtype=np.bool_)),
+]
+
+PydanticNp1DArrayUnicodeString = Annotated[
+    np.ndarray[tuple[int], np.unicode_],
+    NpArrayPydanticAnnotation.factory(
+        data_type=np.unicode_, dimensions=1, strict_data_typing=False
+    ),
+    Field(default_factory=lambda: np.array([], dtype=np.dtype("<U32"))),
+]
+
+
+class SeqTable(PvaTable):
+    repeats: PydanticNp1DArrayInt32
+    trigger: PydanticNp1DArrayUnicodeString
```

Comment:
Can we make this support `Sequence[Enum]` and `Sequence[str]` too please?
```suggestion
    trigger: Sequence[SeqTrigger]
```

Reply:
I've adjusted it so that a `Sequence` of `SeqTrigger` can be given instead. Internally it will always be a unicode string though.